File: version_java/src/main/java/com/example/version_java/data/repo/KakaoRepository.java
package com.example.version_java.data.repo;

import kotlin.jvm.functions.Function1;

public interface KakaoRepository {

    void getSearchList(String location, Function1 callback);
}


//    fun getSearchList(
//            location: String,
//            callback: (result: Result<KakaoSearchResponse>) -> Unit
//        )

File: app/src/main/java/com/example/crimenotification/data/repo/KakaoRepository.kt
package com.example.crimenotification.data.repo

import com.example.crimenotification.network.response.KakaoSearchResponse
import com.example.crimenotification.util.Result

interface KakaoRepository {

    fun getSearchList(
        location: String,
        callback: (result: Result<KakaoSearchResponse>) -> Unit
    )
}